.github/workflows/task-lint.yml

```diff
@@ -0,0 +1,53 @@
+name: Lint for warnings, and check that code is well-formatted
+
+# Documentation: https://redislabs.atlassian.net/wiki/spaces/DX/pages/3967844669/RediSearch+CI+refactor
+
+on:
+  workflow_call:
+    inputs:
+      env:
+        default: "${{ vars.RUNS_ON }}"
+        type: string
+
+env:
+  BOOST_VERSION: ${{ vars.BOOST_VERSION || '1.83.0' }}
+  LLVM_VERSION: ${{ vars.LLVM_VERSION || 18 }}
+
+jobs:
+  lint:
+    name: Linting ${{ inputs.env }}
```

Comment:
Consider removing the env name from the job name, I don't think it's needed there